### ISSUE TITLE
Ping seems to interrupt running video

### ISSUE THREAD

**Issue by haheute** [friendica/friendica#12771]
- [x] I have searched open and closed issues for duplicates

### Bug Description

When I watch a peertube video in friendica, the video stops playing at some time and jumps back to the start of the video.

### Steps to Reproduce

1.  follow a peertube account
2.  watch a video from that account directly in friendica
3.  wait about 30 seconds

Actual Result:

video watching gets interrupted, I believe by the ping function

Expected Result:

Would be nice if the ping would know I am watching and don't do that interruption or .. if the ping had no effect on the playing video.

### Platform Info

Friendica Version: Friendica 'Giant Rhubarb' 2023.03-dev - 1513

PHP version: 8.1.14

SQL version: mysql  Ver 15.1 Distrib 10.6.11-MariaDB, for Linux (x86_64) using readline 5.1

Theme: frio

**Comment by MrPetovan** [comment]
This is likely caused by the “Only refresh on top of page” setting that forces content to be replaced even with no manual interaction (like or comment). I don’t have a good solution for this negative interaction at the moment.

**Comment by haheute** [comment]
okay.. no problem.. maybe later.. fortunately it is possible to watch the videos on the peertube site.
but..
wouldn't it be possible to add a new `<div>click here to show 3 new posts</div>` above the top of the stream? instead of reloading the stream?

**Comment by MrPetovan** [comment]
Yes, but this would require a substantial amount of work to separate this particular from the `ping` + `update_*` system. It looks like I'm the only one that can tackle this long-standing issue and my plate is full with the moderation screens.

**Comment by annando** [comment]
We only should return something with the update mechanism, when there is really an update. This would be a quick step. The rest is more complicate, I think.

**Comment by MrPetovan** [comment]
The setting I mentions adds the “force” parameter to the update calls when at the top of the screen, which always adds content to the update call, even if there’s nothing new. I’m sure there was a good reason at the time to do this, but I can’t currently spend the time to figure out why at the moment.

**Comment by annando** [comment]
Most "good reasons" in the Friendica's UI design are "lack of knowledge" (at least from my side)

**Comment by xundeenergie** [comment]
This Update-mechanism should be realised with websockets... 

On my side, i just made updates of the like/share-buttons with jquery, so you can now like a video or share it, without interrupting watching a video. :)

that was easy. (but needs a little customizing in templates from frio).

The next thing i try is updating the like/share-information below a comment/posting. so i need a little more knowledge about creating templates and new endpoints, i can query with jquery.

and THEN i try websockets for notifications in the navbar.

**Comment by MrPetovan** [comment]
It isn't a question of technology, but of implementation. When we run the update, we don't just update the button that was pressed, but the whole post and its children, which interrupts the playback of any video. We do have an exception in place when something has been typed in a comment field in order not to lose any work this way, but we don't have an exception for video playback.

**Comment by xundeenergie** [comment]
It would be good, to have an interrupt for video-playback too...

**Comment by MrPetovan** [comment]
Maybe we can create a global listener in JQuery for the “play” and “pause” events that would set the same update-pausing flag. It wouldn’t work for iframe-embedded videos though.

**Comment by annando** [comment]
I think the only feasible way is to rework the update mechanism, so that only new and changed parts are updated and that they are only updated when there really had been a change.

**Comment by MrPetovan** [comment]
The “when there really had been a change” is mostly easy to implement, both conversation threads, comments and reactions are objects with a newer date and they impact a different part of the UI. For edits, it’s trickier though.

**Comment by nupplaphil** [comment]
If we really want to update the `ping` / update mechanism, I would like to really separate the API from the Presentation layer.

Means that `ping` shouldn't return anything but native data in JSON format and the presentation layer (= HTML, JS) should render/update it.

**Comment by MrPetovan** [comment]
`/ping` doesn’t return HTML, `/update_*` does.

**Comment by nupplaphil** [comment]
Really? I found at least a Render `notify.tpl` call in a sub call of `Ping.php` .. maybe I was wrong sorry :-/

**Comment by MrPetovan** [comment]
This is only for the notification dropdown menu, it doesn't update the actual content.

**Comment by haheute** [comment]
Now that I post YouTube video links inside BBCode tags like `[video]https://www.youtube....[/video]` , because the preview image for a 'normally' posted YouTube link doesn't appear .. I found this is still a problem. Also for YouTube Videos, not only Peertube.

It looks as if "ping" and then "update_display" stop my running video.. 
:-(

**Comment by haheute** [comment]
okay.. no matter what or why.. I still can't watch a Youtube video in Friendica because it just stops.
Unfortunately I don't have the knowledge to fix that.
Someone who can, please do. thanks

does it work in diaspora? it seems so.
why not in friendica.
